### PR TITLE
🏗  Add script to automatically open and update release tracker issue

### PR DESCRIPTION
**Pull request by estherkim**
TODO: trigger with `release-tagger.yml`

### CODE REVIEW

**Review by festival-up — APPROVED** [review]

File: build-system/release-tagger/update-issue-tracker.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+const promotions = {
+  'beta-opt-in': (tag) =>
+    `- [ ] <!-- tag=${tag} channel=beta-opt-in -->${tag} promoted to Experimental and Beta (opt-in) channels (PROMOTE_TIME)`,
+  'beta-percent': (tag) =>
+    `- [ ] <!-- tag=${tag} channel=beta-percent -->${tag} promoted to Experimental and Beta (1% traffic) channels (PROMOTE_TIME)`,
+  'stable': (tag) =>
+    `- [ ] <!-- tag=${tag} channel=stable -->${tag} promoted to Stable channel (PROMOTE_TIME)`,
+  'lts': (tag) =>
+    `- [ ] <!-- tag=${tag} channel=lts -->(optional) ${tag} promoted to LTS channel (PROMOTE_TIME)`,
+};
```
Comment (danielrozenberg) [inline build-system/release-tagger/update-issue-tracker.js:18]: This feels a bit "magicy" and does some repetition. I think I'd rather this be split into a function and a const that only contains the changing bits - something like:
```suggestion
const CHANNEL_NAMES = {
  'beta-opt-in': 'Experimental and Beta (opt-in) channels',
  'beta-percent': 'Experimental and Beta (1% traffic) channels',
  'stable': 'Stable channel',
  // Not sure how to handle the "optional" part for now :)
  'lts': 'LTS channel',  
};

/**
 * @param {string} channel
 * @param {string} tag
 * @return {string}
 */
function generateTask(channel, tag) {
  return `- [ ] <!-- tag=${tag} channel=${channel} -->${tag} promoted to ${CHANNEL_NAMES[channel]) (PROMOTE_TIME)`;
}
```

File: build-system/release-tagger/test/update-issue-tracker.test.js [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+        '(optional) 2109080123000 promoted to LTS channel (PROMOTE_TIME)\n\n' +
+        '/cc @ampproject/release-on-duty',
+    })
+    .reply(201);
+
+  await createOrUpdateTracker(
+    '2109080123000',
+    '2109010123000',
+    'beta-opt-in',
+    'testtime'
```
[the remark this answers is not in the document]
Comment (estherkim) [reply build-system/release-tagger/test/update-issue-tracker.test.js:47]: Too lazy

File: build-system/release-tagger/utils.js [hunk 3]
```diff
@@ -69,6 +86,23 @@ async function createRelease(tag, commit, body, prerelease) {
...
+async function getIssue(title) {
+  const search = dedent`\
+    search(query:"repo:${owner}/${repo} in:title ${title}", \
+    type: ISSUE, first: 1) { nodes { ... on Issue \
+    { number title body }}}`;
+  const query = `query {${search}}`;
+  const response = await graphqlWithAuth({query});
+  if (response.search?.nodes) {
+    return response.search.nodes[0];
+  }
```
[the remark this answers is not in the document]
Comment (estherkim) [reply build-system/release-tagger/utils.js:103]: Generally would work if `nodes` wasn't an array but getting `[0]` throws an error

File: build-system/release-tagger/update-issue-tracker.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+const promotions = {
+  'beta-opt-in': (tag) =>
+    `- [ ] <!-- tag=${tag} channel=beta-opt-in -->${tag} promoted to Experimental and Beta (opt-in) channels (PROMOTE_TIME)`,
+  'beta-percent': (tag) =>
+    `- [ ] <!-- tag=${tag} channel=beta-percent -->${tag} promoted to Experimental and Beta (1% traffic) channels (PROMOTE_TIME)`,
+  'stable': (tag) =>
+    `- [ ] <!-- tag=${tag} channel=stable -->${tag} promoted to Stable channel (PROMOTE_TIME)`,
+  'lts': (tag) =>
+    `- [ ] <!-- tag=${tag} channel=lts -->(optional) ${tag} promoted to LTS channel (PROMOTE_TIME)`,
```
[the remark this answers is not in the document]
Comment (estherkim) [reply build-system/release-tagger/update-issue-tracker.js]: Good idea! Also helps with the `(optional)` part. I'll follow up with this in another PR.

File: build-system/release-tagger/update-issue-tracker.js (hunk 1 above)
Comment (estherkim) [reply build-system/release-tagger/update-issue-tracker.js]: Good suggestion thank you!

**Review by danielrozenberg — APPROVED** [review]

**Pull request merged**